app/services/by_trust_project_fetcher_service.rb

```diff
@@ -1,38 +1,70 @@
 class ByTrustProjectFetcherService
+  def initialize
+    @all_projects_grouped_by_ukprn = []
+    @all_projects_grouped_by_trust_reference_number = []
+
+    @trusts = []
+    @form_a_mat_trusts = []
+  end
+
   def call
-    projects = projects_by_trust
+    @all_projects_grouped_by_ukprn = projects_grouped_by_ukprn
+    @all_projects_grouped_by_trust_reference_number = projects_grouped_by_trust_reference_number
 
-    if projects
-      trusts = get_trusts(projects.keys)
+    if @all_projects_grouped_by_ukprn.any?
+      @trusts = get_trusts(@all_projects_grouped_by_ukprn.keys)
+    end
 
-      sort_trust_objects_by_name(build_trust_objects(trusts, projects))
-    else
-      []
+    if @all_projects_grouped_by_trust_reference_number.any?
+      @form_a_mat_trusts = build_form_a_mat_trusts
     end
+
+    sort_trust_objects_by_name(build_trust_objects)
+  end
+
+  private def all_trusts
+    @trusts + @form_a_mat_trusts
   end
 
-  private def projects_by_trust
-    projects = Project.active.not_form_a_mat
-    return false unless projects.any?
+  private def all_projects_grouped_by_identifier
+    @all_projects_grouped_by_ukprn.merge(@all_projects_grouped_by_trust_reference_number)
+  end
+
+  private def projects_grouped_by_ukprn
+    Project.active.where(new_trust_reference_number: nil).group_by(&:incoming_trust_ukprn)
```

Comment:
Would it be worth adding a scope for form a mat projects, and not form a mat projects? If you add `Project.form_a_mat` I believe you get `Project.not_form_a_mat` for free. Although might not be worth doing if this is the only place we'll ever use the scopes.

Reply:
`Project.not_form_a_mat` is a scope, but yeah, good shout.

Reply:
> Project.form_a_mat I believe you get Project.not_form_a_mat for free

TIL! ❤️

Reply:
Oh wait! `not_form_a_mat` is a declared scope on Project, the partner `form_a_mat` is not available for free!? :(

Reply:
Oh I'm not sure! I didn't realise there was already a scope! 🤔

Reply:
I've added the scope - scopes are better - thank you!